src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -0,0 +1,313 @@
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class JestElasticsearchClient implements ElasticsearchClient {
+  private static final Logger LOG = LoggerFactory.getLogger(JestElasticsearchClient.class);
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  private final JestClient client;
+  private final Version version;
+
+  @VisibleForTesting
+  public JestElasticsearchClient(String address) {
+    try {
+      JestClientFactory factory = new JestClientFactory();
+      factory.setHttpClientConfig(new HttpClientConfig.Builder(address)
+          .multiThreaded(true)
+          .build()
+      );
+      this.client = factory.getObject();
+      this.version = getServerVersion();
+    } catch (IOException e) {
+      throw new ConnectException(
+          "Couldn't start ElasticsearchSinkTask due to connection error:",
+          e
+      );
+    } catch (ConfigException e) {
+      throw new ConnectException(
+          "Couldn't start ElasticsearchSinkTask due to configuration error:",
+          e
+      );
+    }
+  }
+
+  public JestElasticsearchClient(Map<String, String> props) {
+    try {
+      ElasticsearchSinkConnectorConfig config = new ElasticsearchSinkConnectorConfig(props);
+      final int connTimeout = config.getInt(
+              ElasticsearchSinkConnectorConfig.CONNECTION_TIMEOUT_MS_CONFIG);
+      final int readTimeout = config.getInt(
+              ElasticsearchSinkConnectorConfig.READ_TIMEOUT_MS_CONFIG);
+
+      List<String> address =
+              config.getList(ElasticsearchSinkConnectorConfig.CONNECTION_URL_CONFIG);
+      JestClientFactory factory = new JestClientFactory();
+      factory.setHttpClientConfig(new HttpClientConfig.Builder(address)
+              .connTimeout(connTimeout)
+              .readTimeout(readTimeout)
+              .multiThreaded(true)
+              .build()
+      );
+      this.client = factory.getObject();
+      this.version = getServerVersion();
+    } catch (IOException e) {
+      throw new ConnectException(
+              "Couldn't start ElasticsearchSinkTask due to connection error:",
+              e
+      );
+    } catch (ConfigException e) {
+      throw new ConnectException(
+              "Couldn't start ElasticsearchSinkTask due to configuration error:",
+              e
+      );
+    }
+  }
+
+  private Version getServerVersion() throws IOException {
+    // Default to newest version for forward compatibility
+    Version defaultVersion = Version.SIX;
+
+    NodesInfo info = new NodesInfo.Builder().addCleanApiParameter("version").build();
+    JsonObject result = this.client.execute(info).getJsonObject();
```

Comment:
Again stylistic: Not sure if this was there already, but we tend to use `this` mainly within constructors.

Reply:
I will change it